tests/tx/test_stratum.py

```diff
@@ -38,7 +38,8 @@ def setUp(self):
         super().setUp()
         self.manager = self.create_peer('testnet')
         self.manager.allow_mining_without_peers()
-        self.factory = StratumFactory(self.manager, port=8123, reactor=MemoryReactorHeapClock())
+        port = self.rng.randint(8000, 9000)
```

Comment:
How is this change related to this PR? Is it necessary to be able to run tests simultaneously?

Reply:
Exactly, it's completely unrelated to this PR, but since we've enabled parallel tests this has been happening a lot locally, and _can_ happen on the CI it's just less probable (because it has n=2, but locally it's like n=10).

I didn't think it was worth to open a separate PR, but I could do it.

Reply:
Got it. We can do it here. Just add a note about it in the Acceptance Criteria, please.

Reply:
Another option is changing the `--dist` flag on `pytest-xdist`: https://pytest-xdist.readthedocs.io/en/stable/distribution.html

By doing that we can make all tests in one file, or module, or with the same `@mark`, run in the same worker. That would prevent tests using the same port to run in parallel.

Reply:
> Another option is changing the `--dist` flag on `pytest-xdist`: https://pytest-xdist.readthedocs.io/en/stable/distribution.html
> 
> By doing that we can make all tests in one file, or module, or with the same `@mark`, run in the same worker. That would prevent tests using the same port to run in parallel.

That would be good, I'll just add a note though. I haven't _really_ fixed the problem, just made it much less likely to happen (now a collision needs for both two workers run the test at the same time and get the same port from randint). The `xdist_group` is nice, but it might need a bit of consideration, I think it's still useful for those tests to run in parallel.

Reply:
Also, I suspect that it might be the case for other tests too, any test that listens on a fixed port, or writes to a fixed path is subject to this (I think I got this problem with some resource tests). So we could fix all of those in one go consistently. This was the only one that was happening way too much here.